presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/TypeValidator.java

```diff
@@ -140,11 +147,33 @@ public Void visitUnion(UnionNode node, Void context)
             return null;
         }
 
-        private void checkWindowFunctions(Map<Symbol, WindowNode.Function> functions)
+        private void checkWindowFunctions(Map<Symbol, WindowNode.Function> functions, WindowNode.Specification specification)
         {
+            // Inheritance of the following WindowSpecification components is done by the analyzer. As this information
```

Comment:
I added / changed existing tests in `TestMergeWindows` and `TestReorderWindows` to include the window clauses in the query. Not sure how to add tests in TestMergeAdjacentWindows and TestSwapAdjacentWindowsBySpecifications, since we don't really provide sql statements in the tests, but rather construct `Window` (or now, `WindowInline`) objects.